src/TestSuite/Fixture/FixtureManager.php

```diff
@@ -237,7 +238,7 @@ protected function _setupTable($fixture, $db, array $sources, $drop = true)
         $exists = in_array($table, $sources);
 
         if (($drop && $exists) ||
-            ($exists && !$isFixtureSetup && $fixture->schema() instanceof Table)
+            ($exists && !$isFixtureSetup && $fixture instanceof TableSchemaInterface && $fixture->schema() instanceof Table)
```

Comment:
It certainly would. interface checking are fancier 'method_exists', but in general we are not using them anymore as they don't describe so well what's the method doing.

Also checking for is_subclass_of would work, but I think both solutions are not very timeproof